tools/ingest-converter/src/main/java/org/logstash/ingest/JsUtil.java

```diff
@@ -43,24 +50,50 @@ public static ScriptEngine engine() {
 
     /**
      * Converts the given files from ingest to LS conf using the javascript function
-     * @param args
-     * @param jsFunc
+     * @param args CLI Arguments
+     * @param jsFunc JS function to call
      * @throws ScriptException
      * @throws NoSuchMethodException
      */
-    public static void convert(final String[] args, final String jsFunc) throws ScriptException, NoSuchMethodException {
+    public static void convert(final String[] args, final String jsFunc)
+        throws ScriptException, NoSuchMethodException {
+        final OptionParser parser = new OptionParser();
+        final OptionSpec<URI> input = parser.accepts("input")
+            .withRequiredArg().ofType(URI.class);
+        final OptionSpec<File> output = parser.accepts("output")
+            .withRequiredArg().ofType(File.class);
+        final OptionSet options = parser.parse(args);
         try {
             final ScriptEngine engine = JsUtil.engine();
-            Files.write(Paths.get(args[1]), ((String) ((Invocable) engine).invokeFunction(
-                jsFunc, new String(
-                    Files.readAllBytes(Paths.get(args[0])), StandardCharsets.UTF_8
-                )
-            )).getBytes(StandardCharsets.UTF_8));
+            Files.write(
+                options.valueOf(output).toPath(),
+                ((String) ((Invocable) engine).invokeFunction(
+                    jsFunc, input(options.valueOf(input))
+                )).getBytes(StandardCharsets.UTF_8)
+            );
         } catch (final IOException ex) {
             throw new IllegalStateException(ex);
         }
     }
 
+    /**
+     * Retrieves the input Ingest JSON from a given {@link URI}.
+     * @param uri {@link URI} of Ingest JSON
+     * @return Json String
+     * @throws IOException On failure to load Ingest JSON 
+     */
+    private static String input(final URI uri) throws IOException {
+        if ("file".equals(uri.getScheme())) {
+            return new String(
+                Files.readAllBytes(Paths.get(uri)), StandardCharsets.UTF_8
+            );
+        } else if ("http".equals(uri.getScheme()) || "https".equals(uri.getScheme())) {
+            throw new IllegalArgumentException("http not implemented yet");
+        } else {
+            throw new IllegalArgumentException("--input must be of schema file:// or http(s)://");
```

Comment:
Lets not suggest `http(s)` until we add it :). We can change to `--input must be of schema file://`